### Abstract

The Movie Database (TMDb) is a community built movie and TV database. Every piece of data has been added by our amazing community dating back to 2008. Here, our aim is to extract movie data and reformat it to fit in a Conceptual database schema

### Data Description

The dataset is from Kaggle website. It contains two files. one is tmdb_5000_movies which contains 20 columns and 4803 records and other one is tmdb_5000_credits which contains 4 columns and 4803 records.
Below is the link tot the dataset from Kaggle

#### https://www.kaggle.com/tmdb/tmdb-movie-metadata

In [1]:
#There are two datasets 
#Below are the columns in the tmdb_5000_movies dataset:
#•	budget #Total cost of the movie in which it is made.
#•	genre #Gives information about movie belongs in which all genre
#•	Id #Movie Id which is unique
#•	keywords #Keywords which define movie
#•	original_language #Original language of the movie
#•	overview #Overview of the movie
#•	popularity #Popularity of the movie
#•	production_companies #Company which has produced this movie
#•	production_countries #Countries in which movie is produced
#•	release_date #Date on which movie is released
#•	revenue #Revenue generated by the movie
#•	runtime #Movie’s runtime
#•	tagline #Movie’s tagline
#•	title #Title of the movie
#•	vote_average #Average number of votes movie received
#•	vote_count #Total number of votes movie received

#Below are the columns in the tmdb_5000_credits dataset:
#•	movie_id #Id of the movie
#•	title #Movie title
#•	Cast #Cast of the movie
#•	crew #Crew of the movie

### Importing libraries

In [2]:
import pandas as pd
import numpy as np
import matplotlib as plt
import json
from pandas.io.json import json_normalize

### Reading data from csv

In [3]:
movies_df = pd.read_csv('tmdb_5000_movies.csv')

### Checking output of the data

In [4]:
movies_df.head()

budget                                             genres  \
0  237000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   
1  300000000  [{"id": 12, "name": "Adventure"}, {"id": 14, "...   
2  245000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   
3  250000000  [{"id": 28, "name": "Action"}, {"id": 80, "nam...   
4  260000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   

                                       homepage      id  \
0                   http://www.avatarmovie.com/   19995   
1  http://disney.go.com/disneypictures/pirates/     285   
2   http://www.sonypictures.com/movies/spectre/  206647   
3            http://www.thedarkknightrises.com/   49026   
4          http://movies.disney.com/john-carter   49529   

                                            keywords original_language  \
0  [{"id": 1463, "name": "culture clash"}, {"id":...                en   
1  [{"id": 270, "name": "ocean"}, {"id": 726, "na...                en   
2  [{"id": 470, "name": "spy"}, {"id": 818, "name...                en   
3  [{"id": 849, "name": "dc comics"}, {"id": 853,...                en   
4  [{"id": 818, "name": "based on novel"}, {"id":...                en   

                             original_title  \
0                                    Avatar   
1  Pirates of the Caribbean: At World's End   
2                                   Spectre   
3                     The Dark Knight Rises   
4                               John Carter   

                                            overview  popularity  \
0  In the 22nd century, a paraplegic Marine is di...  150.437577   
1  Captain Barbossa, long believed to be dead, ha...  139.082615   
2  A cryptic message from Bond’s past sends him o...  107.376788   
3  Following the death of District Attorney Harve...  112.312950   
4  John Carter is a war-weary, former military ca...   43.926995   

                                production_companies  \
0  [{"name": "Ingenious Film Partners", "id": 289...   
1  [{"name": "Walt Disney Pictures", "id": 2}, {"...   
2  [{"name": "Columbia Pictures", "id": 5}, {"nam...   
3  [{"name": "Legendary Pictures", "id": 923}, {"...   
4        [{"name": "Walt Disney Pictures", "id": 2}]   

                                production_countries release_date     revenue  \
0  [{"iso_3166_1": "US", "name": "United States o...   2009-12-10  2787965087   
1  [{"iso_3166_1": "US", "name": "United States o...   2007-05-19   961000000   
2  [{"iso_3166_1": "GB", "name": "United Kingdom"...   2015-10-26   880674609   
3  [{"iso_3166_1": "US", "name": "United States o...   2012-07-16  1084939099   
4  [{"iso_3166_1": "US", "name": "United States o...   2012-03-07   284139100   

   runtime                                   spoken_languages    status  \
0    162.0  [{"iso_639_1": "en", "name": "English"}, {"iso...  Released   
1    169.0           [{"iso_639_1": "en", "name": "English"}]  Released   
2    148.0  [{"iso_639_1": "fr", "name": "Fran\u00e7ais"},...  Released   
3    165.0           [{"iso_639_1": "en", "name": "English"}]  Released   
4    132.0           [{"iso_639_1": "en", "name": "English"}]  Released   

                                          tagline  \
0                     Enter the World of Pandora.   
1  At the end of the world, the adventure begins.   
2                           A Plan No One Escapes   
3                                 The Legend Ends   
4            Lost in our world, found in another.   

                                      title  vote_average  vote_count  
0                                    Avatar           7.2       11800  
1  Pirates of the Caribbean: At World's End           6.9        4500  
2                                   Spectre           6.3        4466  
3                     The Dark Knight Rises           7.6        9106  
4                               John Carter           6.1        2124

### Checking shape of data

In [5]:
movies_df.shape

(4803, 20)

In [6]:
movies_df.columns

Index(['budget', 'genres', 'homepage', 'id', 'keywords', 'original_language',
       'original_title', 'overview', 'popularity', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'title', 'vote_average',
       'vote_count'],
      dtype='object')

### Checking data info -  data type and total number of records in each column

In [7]:
movies_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4803 entries, 0 to 4802
Data columns (total 20 columns):
budget                  4803 non-null int64
genres                  4803 non-null object
homepage                1712 non-null object
id                      4803 non-null int64
keywords                4803 non-null object
original_language       4803 non-null object
original_title          4803 non-null object
overview                4800 non-null object
popularity              4803 non-null float64
production_companies    4803 non-null object
production_countries    4803 non-null object
release_date            4802 non-null object
revenue                 4803 non-null int64
runtime                 4801 non-null float64
spoken_languages        4803 non-null object
status                  4803 non-null object
tagline                 3959 non-null object
title                   4803 non-null object
vote_average            4803 non-null float64
vote_count              4803 non-null 

### Checking for null values

In [8]:
movies_df.isnull().sum()

budget                     0
genres                     0
homepage                3091
id                         0
keywords                   0
original_language          0
original_title             0
overview                   3
popularity                 0
production_companies       0
production_countries       0
release_date               1
revenue                    0
runtime                    2
spoken_languages           0
status                     0
tagline                  844
title                      0
vote_average               0
vote_count                 0
dtype: int64

### As homepage contains 3091 null values, removing homepage column

In [9]:
del movies_df['homepage']

#### Checking output - As we can see below homepage column is removed

In [10]:
movies_df.head(2)

budget                                             genres     id  \
0  237000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...  19995   
1  300000000  [{"id": 12, "name": "Adventure"}, {"id": 14, "...    285   

                                            keywords original_language  \
0  [{"id": 1463, "name": "culture clash"}, {"id":...                en   
1  [{"id": 270, "name": "ocean"}, {"id": 726, "na...                en   

                             original_title  \
0                                    Avatar   
1  Pirates of the Caribbean: At World's End   

                                            overview  popularity  \
0  In the 22nd century, a paraplegic Marine is di...  150.437577   
1  Captain Barbossa, long believed to be dead, ha...  139.082615   

                                production_companies  \
0  [{"name": "Ingenious Film Partners", "id": 289...   
1  [{"name": "Walt Disney Pictures", "id": 2}, {"...   

                                production_countries release_date     revenue  \
0  [{"iso_3166_1": "US", "name": "United States o...   2009-12-10  2787965087   
1  [{"iso_3166_1": "US", "name": "United States o...   2007-05-19   961000000   

   runtime                                   spoken_languages    status  \
0    162.0  [{"iso_639_1": "en", "name": "English"}, {"iso...  Released   
1    169.0           [{"iso_639_1": "en", "name": "English"}]  Released   

                                          tagline  \
0                     Enter the World of Pandora.   
1  At the end of the world, the adventure begins.   

                                      title  vote_average  vote_count  
0                                    Avatar           7.2       11800  
1  Pirates of the Caribbean: At World's End           6.9        4500

### Coverting JSON columns to DataFrame

The following example shows a possible JSON representation describing a person.

```json
{
  "firstName": "John",
  "lastName": "Smith",
  "isAlive": true,
  "age": 27,
  "address": {
    "streetAddress": "21 2nd Street",
    "city": "New York",
    "state": "NY",
    "postalCode": "10021-3100"
  },
  "phoneNumbers": [
    {
      "type": "home",
      "number": "212 555-1234"
    },
    {
      "type": "office",
      "number": "646 555-4567"
    },
    {
      "type": "mobile",
      "number": "123 456-7890"
    }
  ],
  "children": [],
  "spouse": null
}
```


#### Below are the steps to convert json object to dataframe

#### First converting genre json column into dataframe

#### step 1 : Create a empty list. Below I have created empty list  to collect genre data into "genre_column_list"

In [11]:
genre_column_list =[]

#### step 2 : Appending genre json data into python list "genre_column_list". Using json.loads() method to parse data

In [12]:
for each in movies_df['genres']:
    genre_column_list.append(json.loads(each))

#### Checking result in genre_column_list by displaying 5 records

In [13]:
genre_column_list[:5]

[[{'id': 28, 'name': 'Action'},
  {'id': 12, 'name': 'Adventure'},
  {'id': 14, 'name': 'Fantasy'},
  {'id': 878, 'name': 'Science Fiction'}],
 [{'id': 12, 'name': 'Adventure'},
  {'id': 14, 'name': 'Fantasy'},
  {'id': 28, 'name': 'Action'}],
 [{'id': 28, 'name': 'Action'},
  {'id': 12, 'name': 'Adventure'},
  {'id': 80, 'name': 'Crime'}],
 [{'id': 28, 'name': 'Action'},
  {'id': 80, 'name': 'Crime'},
  {'id': 18, 'name': 'Drama'},
  {'id': 53, 'name': 'Thriller'}],
 [{'id': 28, 'name': 'Action'},
  {'id': 12, 'name': 'Adventure'},
  {'id': 878, 'name': 'Science Fiction'}]]

#### step 3 : Create empty list to collect movieid

In [14]:
movieid_genre_list = []

#### step 4 : Here, we are using built-in function enumerate which adds a counter to genere_column_list and returns enumerat object.
#### This enumerate object is used directly in for loop which returns a list of movieid's

In [15]:
for i,x in enumerate(genre_column_list):
    for y in x:
        movieid_genre_list.append(movies_df['id'][i])

#### Below I am displaying 5 records in movieid_genre_list 

In [16]:
movieid_genre_list[:5]

[19995, 19995, 19995, 19995, 285]

#### step 5 : Creating genre_list and genre_id_list to collect genre id and name which are the records of genre column

In [17]:
genre_list = []
genre_id_list=[]

#### step 6 : Using for loop to append name and id in genre_list and genre_id_list respectively

In [18]:
for x in genre_column_list:
    for y in x:
        genre_id_list.append(y['id'])
        genre_list.append(y['name'])           

#### Displaying 5 records in genre_list 

In [19]:
genre_list[:5]

['Action', 'Adventure', 'Fantasy', 'Science Fiction', 'Adventure']

In [20]:
genre_id_list[:5]

[28, 12, 14, 878, 12]

#### step 6 : Creating dataframe using pandas library which contains movie_id, genre_id and genre

In [21]:
genre_df = pd.DataFrame(genre_list, columns=['Genre'])
genre_df['ID'] = pd.Series(genre_id_list)
genre_df['Movie_ID'] = pd.Series(movieid_genre_list)
genre_df = genre_df[['Movie_ID','ID', 'Genre']]

#### Below is the output displaying few records

In [22]:
genre_df.head()

Movie_ID   ID            Genre
0     19995   28           Action
1     19995   12        Adventure
2     19995   14          Fantasy
3     19995  878  Science Fiction
4       285   12        Adventure

#### Similarly, Converting production companies column which contains json object into dataframe. Here, Also, we will perform same steps which are mentioned above.

#### step 1 : Create empty list to hold productioncomp data

In [23]:
productioncomp_column_list = []

#### step 2 : parsing json object into python list using json.loads() method

In [24]:
for each in movies_df['production_companies']:
    productioncomp_column_list.append(json.loads(each))

#### Displaying ouput by showing 5 records

In [25]:
productioncomp_column_list[:5]

[[{'id': 289, 'name': 'Ingenious Film Partners'},
  {'id': 306, 'name': 'Twentieth Century Fox Film Corporation'},
  {'id': 444, 'name': 'Dune Entertainment'},
  {'id': 574, 'name': 'Lightstorm Entertainment'}],
 [{'id': 2, 'name': 'Walt Disney Pictures'},
  {'id': 130, 'name': 'Jerry Bruckheimer Films'},
  {'id': 19936, 'name': 'Second Mate Productions'}],
 [{'id': 5, 'name': 'Columbia Pictures'},
  {'id': 10761, 'name': 'Danjaq'},
  {'id': 69434, 'name': 'B24'}],
 [{'id': 923, 'name': 'Legendary Pictures'},
  {'id': 6194, 'name': 'Warner Bros.'},
  {'id': 9993, 'name': 'DC Entertainment'},
  {'id': 9996, 'name': 'Syncopy'}],
 [{'id': 2, 'name': 'Walt Disney Pictures'}]]

#### step 3 : Creating two empty list to hold name and id

In [26]:
company_id_list = []
company_list=[]

#### step 4 : Using for loop extracting id and name and storing it in company_id_list and company_list

In [27]:
for x in productioncomp_column_list:
    for y in x:
        company_id_list.append(y['id'])
        company_list.append(y['name'])

#### step 5 : Creating empyt list to store movieid

In [28]:
movieid_prodcomp_list = []

#### step 6 : Here, we are using built-in function enumerate which adds a counter to productioncomp_column_list and returns enumerate object.
#### This enumerate object is used directly in for loop which returns a list of movieid's

In [29]:
for i, x in enumerate(productioncomp_column_list):
    for y in x:
        movieid_prodcomp_list.append(movies_df['id'][i])

#### Displaying results

In [30]:
movieid_prodcomp_list[:10]

[19995, 19995, 19995, 19995, 285, 285, 285, 206647, 206647, 206647]

#### step 7 : Creating dataframe for production companies

In [31]:
prod_companies_df = pd.DataFrame(movieid_prodcomp_list, columns = ['Movie_ID'])
prod_companies_df['ID'] = pd.Series(company_id_list)
prod_companies_df['Company'] = pd.Series(company_list)

#### Displaying result

In [32]:
prod_companies_df[:10]

Movie_ID     ID                                 Company
0     19995    289                 Ingenious Film Partners
1     19995    306  Twentieth Century Fox Film Corporation
2     19995    444                      Dune Entertainment
3     19995    574                Lightstorm Entertainment
4       285      2                    Walt Disney Pictures
5       285    130                 Jerry Bruckheimer Films
6       285  19936                 Second Mate Productions
7    206647      5                       Columbia Pictures
8    206647  10761                                  Danjaq
9    206647  69434                                     B24

#### Reading data from second database cast

This dataset contains information about cast of the movie

#### Reading tmdb_5000_credits.csv file using pandas library

In [33]:
cast_xl_df = pd.read_csv('tmdb_5000_credits.csv')

#### Displaying output

In [34]:
cast_xl_df.head()

movie_id                                     title  \
0     19995                                    Avatar   
1       285  Pirates of the Caribbean: At World's End   
2    206647                                   Spectre   
3     49026                     The Dark Knight Rises   
4     49529                               John Carter   

                                                cast  \
0  [{"cast_id": 242, "character": "Jake Sully", "...   
1  [{"cast_id": 4, "character": "Captain Jack Spa...   
2  [{"cast_id": 1, "character": "James Bond", "cr...   
3  [{"cast_id": 2, "character": "Bruce Wayne / Ba...   
4  [{"cast_id": 5, "character": "John Carter", "c...   

                                                crew  
0  [{"credit_id": "52fe48009251416c750aca23", "de...  
1  [{"credit_id": "52fe4232c3a36847f800b579", "de...  
2  [{"credit_id": "54805967c3a36829b5002c41", "de...  
3  [{"credit_id": "52fe4781c3a36847f81398c3", "de...  
4  [{"credit_id": "52fe479ac3a36847f813eaa3", "de...

#### Similarly, lets extract "cast" json object into dataframe

#### Step 1 - Create Empty list to store cast column

In [35]:
cast_column_list = []

#### Step 2 - Appending cast json data into python list "cast_list". Using json.loads() method to parse data 

In [36]:
for each in cast_xl_df['cast']:
    cast_column_list.append(json.loads(each))

#### Checking result in genre_column_list by displaying 5 records

In [37]:
cast_column_list[:5]

[[{'cast_id': 242,
   'character': 'Jake Sully',
   'credit_id': '5602a8a7c3a3685532001c9a',
   'gender': 2,
   'id': 65731,
   'name': 'Sam Worthington',
   'order': 0},
  {'cast_id': 3,
   'character': 'Neytiri',
   'credit_id': '52fe48009251416c750ac9cb',
   'gender': 1,
   'id': 8691,
   'name': 'Zoe Saldana',
   'order': 1},
  {'cast_id': 25,
   'character': 'Dr. Grace Augustine',
   'credit_id': '52fe48009251416c750aca39',
   'gender': 1,
   'id': 10205,
   'name': 'Sigourney Weaver',
   'order': 2},
  {'cast_id': 4,
   'character': 'Col. Quaritch',
   'credit_id': '52fe48009251416c750ac9cf',
   'gender': 2,
   'id': 32747,
   'name': 'Stephen Lang',
   'order': 3},
  {'cast_id': 5,
   'character': 'Trudy Chacon',
   'credit_id': '52fe48009251416c750ac9d3',
   'gender': 1,
   'id': 17647,
   'name': 'Michelle Rodriguez',
   'order': 4},
  {'cast_id': 8,
   'character': 'Selfridge',
   'credit_id': '52fe48009251416c750ac9e1',
   'gender': 2,
   'id': 1771,
   'name': 'Giovanni Rib

#### step 3 : Create empty list to collect movieid¶

In [38]:
movieid_cast_list = []

#### step 4 : Here, we are using built-in function enumerate which adds a counter to cast_column_list and returns enumerat object.This enumerate object is used directly in for loop which returns a list of movieid's

In [39]:
for i,x in enumerate(cast_column_list):
    for y in x:
        movieid_cast_list.append(movies_df['id'][i])

#### Below I am displaying 5 records in movieid_genre_list

In [40]:
movieid_cast_list[:5]

[19995, 19995, 19995, 19995, 19995]

#### step 5 : Creating list for all the elements in cast column

In [41]:
castid_list = []
character_list = []
creditid_list = []
gender_list = []
id_list = []
name_list = []
order_list = []

#### step 6 : Using for loop to append all the items respectively

In [42]:
for x in cast_column_list:
    for y in x:
        castid_list.append(y['cast_id'])
        character_list.append(y['character'])
        creditid_list.append(y['credit_id'])
        gender_list.append(y['gender'])
        id_list.append(y['id'])
        name_list.append(y['name'])
        order_list.append(y['order'])

#### Displaying 5 records in genre_list 

In [43]:
len(castid_list)

106257

In [44]:
len(character_list)

106257

In [45]:
len(creditid_list)

106257

In [46]:
len(gender_list)

106257

In [47]:
len(id_list)

106257

In [48]:
len(name_list)

106257

In [49]:
len(order_list)

106257

#### step 6 : Creating dataframe using pandas library which contains movie_id, genre_id and genre

In [50]:
df = pd.DataFrame()

In [51]:
df['Cast_ID'] = castid_list

In [52]:
df['Character'] = character_list

In [53]:
df['Credit_ID'] = creditid_list

In [54]:
df['Gender'] = gender_list

In [55]:
df['ID'] = id_list

In [56]:
df['Name'] = name_list

In [57]:
df['Order'] = order_list

In [58]:
df['MovieID'] = movieid_cast_list

In [59]:
df.head()

Cast_ID            Character                 Credit_ID  Gender     ID  \
0      242           Jake Sully  5602a8a7c3a3685532001c9a       2  65731   
1        3              Neytiri  52fe48009251416c750ac9cb       1   8691   
2       25  Dr. Grace Augustine  52fe48009251416c750aca39       1  10205   
3        4        Col. Quaritch  52fe48009251416c750ac9cf       2  32747   
4        5         Trudy Chacon  52fe48009251416c750ac9d3       1  17647   

                 Name  Order  MovieID  
0     Sam Worthington      0    19995  
1         Zoe Saldana      1    19995  
2    Sigourney Weaver      2    19995  
3        Stephen Lang      3    19995  
4  Michelle Rodriguez      4    19995

### Final CAST and MOVIES dataframe:

In [60]:
cast_df = pd.DataFrame(movieid_cast_list, columns=['Movie_ID'])
cast_df['ID'] = pd.Series(castid_list)
cast_df['Character'] = pd.Series(character_list)
cast_df['Actor'] = pd.Series(name_list)

#### Displaying output

In [61]:
cast_df.head()

Movie_ID   ID            Character               Actor
0     19995  242           Jake Sully     Sam Worthington
1     19995    3              Neytiri         Zoe Saldana
2     19995   25  Dr. Grace Augustine    Sigourney Weaver
3     19995    4        Col. Quaritch        Stephen Lang
4     19995    5         Trudy Chacon  Michelle Rodriguez

#### step 1 : Create a empty list. Below I have created empty list to collect countries data into "countries_column_list"

In [62]:
countries_column_list =[]

#### step 2 : Appending production_countries json data into python list "countries_column_list". Using json.loads() method to parse data

In [63]:
for each in movies_df['production_countries']:
    countries_column_list.append(json.loads(each))

#### Checking result in genre_column_list by displaying 5 records

In [64]:
countries_column_list[:5]

[[{'iso_3166_1': 'US', 'name': 'United States of America'},
  {'iso_3166_1': 'GB', 'name': 'United Kingdom'}],
 [{'iso_3166_1': 'US', 'name': 'United States of America'}],
 [{'iso_3166_1': 'GB', 'name': 'United Kingdom'},
  {'iso_3166_1': 'US', 'name': 'United States of America'}],
 [{'iso_3166_1': 'US', 'name': 'United States of America'}],
 [{'iso_3166_1': 'US', 'name': 'United States of America'}]]

#### step 3 : Create empty list to collect movieid

In [65]:
movieid_countries_list = []

#### step 4 : Here, we are using built-in function enumerate which adds a counter to countries_column_list and returns enumerat object.This enumerate object is used directly in for loop which returns a list of movieid's

In [66]:
for i,x in enumerate(countries_column_list):
    for y in x:
        movieid_countries_list.append(movies_df['id'][i])

#### Below I am displaying 5 records in movieid_genre_list

In [67]:
movieid_countries_list[:5]

[19995, 19995, 285, 206647, 206647]

#### step 5 : Creating list to collect countries_name_list

In [68]:
countries_name_list=[]

#### step 6 : Using for loop to append name in name list respectively

In [69]:
for x in countries_column_list:
    for y in x:
        countries_name_list.append(y['name'])

#### Displaying 5 records in genre_list 

In [70]:
countries_name_list[:5]

['United States of America',
 'United Kingdom',
 'United States of America',
 'United Kingdom',
 'United States of America']

#### step 6 : Creating dataframe using pandas to display movie and location in which movie is shooted

In [71]:
countries_df = pd.DataFrame(countries_name_list, columns=['Shoot_Location'])

In [72]:
countries_df['Movie_ID'] = pd.Series(movieid_countries_list)

In [73]:
countries_df = countries_df[['Movie_ID', 'Shoot_Location']]

In [74]:
countries_df.head()

Movie_ID            Shoot_Location
0     19995  United States of America
1     19995            United Kingdom
2       285  United States of America
3    206647            United Kingdom
4    206647  United States of America

#### Collecting movies which are having vote avergae greater than 7

In [75]:
movieid_abv_7 = movies_df[movies_df['vote_average'] >= 7]['id']

In [76]:
movieid_abv_7 = list(movieid_abv_7)

In [77]:
movieid_abv_7[:5]

[19995, 49026, 38757, 99861, 767]

#### Checking for actors having vote_average greater than 7

In [78]:
actors_abv_7 = []

In [79]:
for i, x in enumerate(cast_df['Movie_ID']):
    if x in movieid_abv_7:
        actors_abv_7.append(cast_df['Actor'][i])

In [80]:
actors_abv_7 = list(set(actors_abv_7))

In [81]:
indices = [55,122,116,117,118]

In [82]:
for i in sorted(indices, reverse=True):
    del actors_abv_7[i]

In [83]:
len(actors_abv_7)

21707

### License

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

MIT License https://opensource.org/licenses/MIT